groovy-language-server/src/main/java/com/palantir/ls/server/groovy/GroovyWorkspaceCompiler.java

```diff
@@ -227,17 +228,19 @@ private void addAllSourcesToCompilationUnit() {
     private void resetCompilationUnit() {
         try {
             FileUtils.deleteDirectory(config.getTargetDirectory());
-            if (!config.getTargetDirectory().mkdir()) {
-                logger.error("Could not recreate target directory: '{}'",
-                        config.getTargetDirectory().getAbsolutePath());
-                throw new RuntimeException("Could not reset compiled files after changes. "
-                        + "Make sure you have permission to modify your target directory.");
-            }
-            unit = new CompilationUnit(config);
-            addAllSourcesToCompilationUnit();
         } catch (IOException e) {
-            Throwables.propagate(e);
+            logger.error("Could not delete directory '" + config.getTargetDirectory().toString() + "'", e);
+            throw new RuntimeException("Could not reset compiled files after changes. "
+                    + "Make sure you have permission to modify your target directory.");
+        }
+        if (!config.getTargetDirectory().mkdir()) {
+            logger.error("Could not recreate target directory: '{}'",
+                    config.getTargetDirectory().getAbsolutePath());
+            throw new RuntimeException("Could not reset compiled files after changes. "
+                    + "Make sure you have permission to modify your target directory.");
```

Comment:
After talking in person we decided on: `User may not have permission to modify target directory.` since `GroovyWorkspaceCompiler` could be given a non-tmp directory.